src/utils/proofs_api/paramfetch.rs

```diff
@@ -171,53 +169,48 @@ async fn fetch_verify_params(
 
     fetch_params(&path, &info).await?;
 
-    check_file(path, info).await.map_err(|e| {
+    check_file(&path, &info).await.map_err(|e| {
         // TODO remove invalid file
         e.into()
     })
 }
 
-async fn fetch_params(path: &Path, info: &ParameterData) -> Result<(), anyhow::Error> {
+async fn fetch_params(path: &Path, info: &ParameterData) -> anyhow::Result<()> {
+    let cid = Cid::from_str(&info.cid)?;
     let gw = std::env::var(GATEWAY_ENV).unwrap_or_else(|_| GATEWAY.to_owned());
     info!("Fetching param file {:?} from {}", path, gw);
-    let url = format!("{}{}", gw, info.cid);
     let result = retry(ExponentialBackoff::default(), || async {
-        Ok(fetch_params_inner(&url, path).await?)
+        Ok(async {
+            let tmp =
+                tempfile::NamedTempFile::new_in(path.parent().unwrap_or_else(|| Path::new(".")))?
+                    .into_temp_path();
+            download_ipfs_file_trustlessly(&cid, Some(GATEWAY), &tmp).await?;
+            // The extra check is redundant, just making sure `download_ipfs_file_trustlessly` and `rs-car-ipfs` is bug free, consider removing it in the future.
+            check_file(&tmp, info).await?;
```

Comment:
I'm good with removing this.

Reply:
How costly is this check? Is it valuable as a final sanity check?

Reply:
It's pretty cheap for key files given their sizes are mostly a few KB

Reply:
Removed, the check will still be performed on every start of the process to validate local param cache